internal/wal/wal_aof.go

```
@@ -65,31 +170,212 @@ func (w *WALAOF) LogCommand(c *cmd.DiceDBCmd) {
 	data, err := proto.Marshal(entry)
 	if err != nil {
 		slog.Warn("failed to serialize command", slog.Any("error", err.Error()))
+		return
```

We shouldn't silently ignore these errors. They should be returned to caller.